### PR TITLE
enable multiple environments

### PR DESCRIPTION
**Pull request by gitteri**
Allow for multiple vault environments to be stored and used independently.
Also default to VAULT_TOKEN env var as token if it is available

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -64,6 +62,17 @@ func realMain() error {
 // an error, instead the empty string is returned. Errors are returned if there
 // are communication issues with the keychain itself.
 func handleGet() error {
+	// if there is a vault token stored as an env var use it first
+	envToken := os.Getenv("VAULT_TOKEN")
```
Comment (gumdelli) [inline main.go]: Do we ever store the vault token as an env variable?

File: main.go [hunk 2]
```diff
@@ -64,6 +62,17 @@ func realMain() error {
...
+	if envToken != "" {
+		_, err := fmt.Fprintf(stdout, "%s", envToken)
+		if err != nil {
+			return errors.Wrap(err, "failed to print vault token from env var")
+		}
+		return nil
+	}
+
+	// otherwise use the keychain
```
Comment (gumdelli) [inline main.go]: Also a side comment: can we use an os agnostic keyring like [this one](https://github.com/zalando/go-keyring)? We might be able to use this for our servers too.

File: main.go (hunk 1 above)
Comment (gitteri) [reply main.go]: For running certain applications locally (cryptocoin service, pax etc) we do use the convenience of setting a vault token locally to give us quick access to the local vault in k8

File: main.go (hunk 2 above)
Comment (jgiles) [reply main.go]: worth considering. that also sounds like we might be able to avoid cgo

Comment (jgiles) [reply main.go]: actually it looks like the lib used here claims to support linux too (https://github.com/keybase/go-keychain), but the readme only discusses osx/ios?

File: main.go [hunk 3]
```diff
@@ -64,6 +62,17 @@ func realMain() error {
 // an error, instead the empty string is returned. Errors are returned if there
 // are communication issues with the keychain itself.
 func handleGet() error {
+	// if there is a vault token stored as an env var use it first
+	envToken := os.Getenv("VAULT_TOKEN")
+	if envToken != "" {
+		_, err := fmt.Fprintf(stdout, "%s", envToken)
+		if err != nil {
+			return errors.Wrap(err, "failed to print vault token from env var")
```
Comment (jgiles) [inline main.go]: I'd also be content to just `fmt.Printf` here rather than `fmt.Fprintf(os.stdout`, and avoid needing to deal with the error return value

Comment (jgiles) [reply main.go]: similarly, the only place we appear to use `stderr` is to print the single error message. Could just use the Go default `log` package there (which prints to stderr)

**Pull request merged**